assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -459,117 +353,43 @@ fn handle_division(
 /// Mutate the first two elements of the stack from `[b, a, ..]` into `[2^b, a, ..]`, with `b`
 /// either as a provided immediate value, or as an element that already exists in the stack.
 ///
-/// If the used mode is `checked`, the function will assert that both `[b, a]` are valid `u32`.
-/// This function is equivalent to a bit shift operation, so the exponent shouldn't cause a number
-/// to be greater than `u32::MAX`; therefore, the maximum valid value must be `31`, as defined in
-/// the helper constants.
-///
-/// This function supports only checked and unchecked modes; if some other mode is provided, it
-/// will panic.
+/// This function supports only unchecked mode; if some other mode is provided, it will panic.
 fn prepare_bitwise<const MAX_VALUE: u8>(
     span: &mut SpanBuilder,
     imm: Option<u8>,
-    op_mode: U32OpMode,
     final_ops: [Operation; 2],
```

Comment:
I would probably also remove `final_ops` from here and move their handling to the respective function. This would be more consistent with how we use `handle_division`.